app/javascript/packs/unpublishPostModal.jsx

```diff
@@ -30,127 +28,60 @@ async function confirmAdminUnpublishPost(id, username, slug) {
     });
   }
 
-  toggleUnpublishPostModal();
+  closeModal();
 }
 
-/**
- * Shows or hides the flag user modal.
- */
-export function toggleUnpublishPostModal() {
-  const modalContainer = top.document.getElementsByClassName(
-    'unpublish-post-modal-container',
-  )[0];
-  modalContainer.classList.toggle('hidden');
-
-  if (!modalContainer.classList.contains('hidden')) {
-    top.window.scrollTo(0, 0);
-    top.document.body.style.height = '100vh';
-    top.document.body.style.overflowY = 'hidden';
-  } else {
-    top.document.body.style.height = 'inherit';
-    top.document.body.style.overflowY = 'inherit';
-  }
+function closeModal() {
+  closeWindowModal(window.parent.document);
 }
 
+const modalContents = new Map();
```

Comment:
I think this can be simpler - in other places in the app we use a `Map` like this, because we expect to hold and control multiple different modals. But in this case we only ever hold/control the unpublish post modal (I'm pretty sure).

So instead of holding a `Map`, we could just have e.g.

```js
let modalContents;
```

and then below in `getModalContents` we could do:

```js
  if (!modalContents) {

    const modalContentElement =
      window.parent.document.querySelector(modalContentSelector);
    modalContents = modalContentElement.innerHTML;
    modalContentElement.remove();
  }
  return modalContents;
```

Reply:
Makes sense. Applied these changes. Thanks.